packages/next-admin/src/utils/tools.ts

```diff
@@ -20,9 +20,26 @@ export const isScalar = (value: string | boolean | number) => {
   );
 };
 
-
-export const pipe = <T>(...fns: Function[]) =>  (x: T) => {
+export const pipe = <T>(...fns: Function[]) => (x: T) => {
   return fns.reduce(async (v, f) => {
     return await f(await v)
   }, Promise.resolve(x));
-}
\ No newline at end of file
+}
+
+export const extract = <T>(obj: T, prototype: Partial<T>): T => {
```

Comment:
What is the purpose of this function (add a comment maybe ?)
Also maybe add a test